src/utils/bzz-address.ts

```diff
@@ -0,0 +1,18 @@
+export class BzzAddress {
+  public hash: string
+  public path: string | null
+
+  constructor(url: string) {
+    if (url.startsWith('bzz://')) {
+      url = url.slice(6)
+    }
+    const parts = url.split('/')
+    this.hash = parts[0].toLowerCase()
+
+    if (!/[a-z0-9]{64,128}/i.test(this.hash)) {
+      throw new Error('Invalid BZZ address: expected 64 or 128 long hexadecimal hash')
+    }
+    const pathParts = parts.slice(1).filter(x => x)
```

Comment:
The mantaray structure allows `//` path in general, but on filesystem level and usage it would be better maybe to just throw an error instead.